cli/azd/pkg/commands/pipeline/github_provider.go

```diff
@@ -358,17 +364,73 @@ func (p *GitHubCiProvider) configureConnection(
 	repoDetails *gitRepositoryDetails,
 	infraOptions provisioning.Options,
 	credentials json.RawMessage,
+	authType PipelineAuthType,
 	console input.Console) error {
 
+	// Federated Auth + Terraform is not a supported combination
+	if infraOptions.Provider == provisioning.Terraform {
+		// Throw error if Federated auth is explicitly requested
+		if authType == AuthTypeFederated {
+			return fmt.Errorf(
+				//nolint:lll
+				"Terraform does not support federated authentication. Service Principal with client ID and client secret must be used. %w",
```

Comment:
How about saying if they want to explicitly set it then which param to use, like --auth-type client-credentials.